From 1f8ba33c0fcca2f9a8382e1a8feb1e3ea78cc582 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Tue, 19 May 2015 11:36:58 +0800
Subject: [PATCH] anaconda: delete environment TERMCAP

While using screen for text display, "TERMCAP" was set by screen,
it caused 'Aborted' failure if multiple SnackScreen instances was
generated, so delete it to avoid the issue.

How to reproduce the issue in an easy way:
1) Start a terminal(such as ssh/serial/tty2), and run screen
$ screen

2) Run the following python script, it aborted with segment fault.
$ cat > test.py <<ENDOF
from snack import *

screen = SnackScreen()
screen.finish()

screen = SnackScreen()
screen.finish()
ENDOF

$ python test.py
Segmentation fault

BTW, if you directly run "screen python test.py", everything is ok,
so the normal anaconda starting will not be affected, it occurs only
if manually stop/start anaconda in a screen session.

Upstream-Status: Inappropriate [WRLinux specific]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 anaconda | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/anaconda b/anaconda
index 7191b50..fc2d3fb 100755
--- a/anaconda
+++ b/anaconda
@@ -256,6 +256,12 @@ def setupEnvironment():
     os.environ["GLADEPATH"] = "/tmp/updates/:/tmp/updates/data/ui/:ui/:/usr/share/anaconda/ui/:/usr/share/python-meh/"
     os.environ["PIXMAPPATH"] = "/tmp/updates/pixmaps/:/tmp/updates/:/tmp/product/pixmaps/:/tmp/product/:pixmaps/:/usr/share/anaconda/pixmaps/:/usr/share/pixmaps/:/usr/share/anaconda/:/usr/share/python-meh/:/usr/share/icons/Fedora/48x48/apps/"
 
+    # While using screen for text display, "TERMCAP" was set by screen,
+    # it caused 'Aborted' failure if multiple SnackScreen instances was
+    # generated, so delete it to avoid the issue
+    if "TERMCAP" in os.environ.keys():
+        del os.environ["TERMCAP"]
+
 def setupLoggingFromOpts(opts):
     if opts.loglevel and anaconda_log.logLevelMap.has_key(opts.loglevel):
         level = anaconda_log.logLevelMap[opts.loglevel]
-- 
1.9.1

